File: ICPCCommunity/Sheet4/ValeraAndX.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package icpcsheet4;

import java.util.Scanner;

/**
 *
 * @author amirh
 */
public class ValeraAndX {
    
     public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        char [][] ch = new char[n][n];
        char c;
        boolean con = true;
         for (int i = 0; i < ch.length; i++) {
             for (int j = 0; j < ch.length; j++) {
                 ch[i][j] = in.next().charAt(j);
             }
             
         }
         
         c = ch[0][0];
         for (int i = 0; i < n; i++) {
             if(ch[i][i] != c){
                 con = false;
                 break;
             }
         }
         for (int i = 0 ; i <n; i++) {
             if(ch[i][n-1-i] != c){
                 con = false;
                 break;
             }
         }
             
         if(con){
             System.out.println("YES");
             
         }
         else
             System.out.println("NO");
         
                
     }
}

File: CodeForces/Problem A/Valera and X.cpp
#include <iostream>
#include <set>
using namespace std;

int main()
{
    set<char> ss;
    int n;
    cin>>n;
    bool con = true;
    char ch[n][n];
    for(int i =0;i<n;i++){
        for(int j =0;j<n;j++){
            cin>>ch[i][j];
            ss.insert(ch[i][j]);
        }
    }
    if(ss.size() !=2){
        cout<<"NO"<<endl;
        return 0;
    }
    char c = ch[0][0];
    char y = ch[0][1];
    for (int i = 0; i < n; i++) {
             if(ch[i][i] != c){
                 con = false;
                 break;
             }
         }

    for (int i = 0 ; i <n; i++) {
             if(ch[i][i] != c || ch[i][n-(i+1)] != c || ch[i][i] == y || ch[i][n-(i+1)] == y  ){
                 con = false;
                 break;
             }
             for(int j = 0; j < n; j++){
            if(i != j && j != (n - (i + 1)))
            {
                if(ch[i][j] != y){
                    con = false;
                    break;
                }
            }
        }
         }



    if(con)
        cout<<"YES"<<endl;
    else
        cout<<"NO"<<endl;
    return 0;
}
